mev/poll.go

```diff
@@ -0,0 +1,69 @@
+package mev
+
+import (
+	"context"
+	types "github.com/SiloMEV/silo-mev-protobuf-go/mev/v1"
+	"github.com/tendermint/tendermint/libs/log"
+	"google.golang.org/grpc"
+	"time"
+)
+
+type Poller struct {
+	client            types.BundleProviderClient
+	keeper            *Keeper
+	lastBlockProvider func() int64
+	logger            log.Logger
+	ctx               context.Context
+}
+
+func (p *Poller) run() {
+
+	lastHeight := p.lastBlockProvider()
+
+	bundles, err := p.client.GetBundles(context.Background(), &types.GetBundlesRequest{MinBlockHeight: uint64(lastHeight)})
+	if err != nil {
+		p.logger.Error("Error while querying SILO server for bundles", "err", err)
+		return
+	}
+	// TODO validate data, don't trust height
+	for height, bundles := range bundles.Bundles {
+		p.keeper.AddBundles(int64(height), bundles.Bundles)
+	}
+
+}
+
+func NewPoller(logger log.Logger, config Config, keeper *Keeper, ctx context.Context, lastBlockProvider func() int64) (*Poller, error) {
+
+	logger.Info("Starting bundle provider poller")
+
+	// TODO security options?
+	grpcConn, err := grpc.DialContext(context.Background(), config.ServerAddr, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+
+	bundleProviderClient := types.NewBundleProviderClient(grpcConn)
+
+	p := &Poller{
+		client:            bundleProviderClient,
+		keeper:            keeper,
+		lastBlockProvider: lastBlockProvider,
+		logger:            logger,
+		ctx:               ctx,
+	}
+
+	ticker := time.NewTicker(config.PollInterval)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				p.run()
```

Comment:
should you reset the ticker after this? if the runs take a measurable amount of time and there's no back pressure, validators/nodes could end up crashing the mev server.

Reply:
Interesting point. I was under the same impression initially, but turns out `NewTicker` docs says `The ticker will adjust the time interval or drop ticks to make up for slow receivers.`. From what I gather this means it won't fire new ticks until the last one is read (also `.C` channel has a size of 1). 
So it won't create multiple calls cascade if they take more then `pollInterval`.

Reply:
sounds good!